### ISSUE TITLE
TROCR custom dataset

### ISSUE THREAD

**Issue by Jim0530** [huggingface/transformers#14248]
Hi everyone 
I would like to train a visionencoderdecoder on a custom dataset
Which I hope my decoder can decode different  words from custom vocabulary
How can I train my own model without using any pretrained model
Thanks

**Comment by NielsRogge** [comment]
Hi!

Thanks for your interest in using `VisionEncoderDecoderModel`.

The steps to proceed would be:
1. train a new tokenizer on your custom text data, using [HuggingFace Tokenizers](https://github.com/huggingface/tokenizers). Could be a BPE (byte pair encoding) tokenizer for instance. Next, initialize your token with the new vocabulary.
```
tokenizer = ...
```
2. choose a text model to use as decoder, e.g. BERT, GPT2, RoBERTA. Let's choose BERT: 
```
from transformers import BertConfig, BertModel

decoder_config = BertConfig()
decoder = BertModel(decoder_config)
```
3. make sure that the embedding layer of the Transformer that you'll use as decoder has the same length as your newly trained tokenizer, i.e run the following:
```
decoder.resize_token_embeddings(len(tokenizer))
```
4. initialize a `VisionEncoderDecoderModel` with an image encoder (e.g. ViT) and your new decoder
```
from transformers import ViTConfig, ViTModel, VisionEncoderDecoderModel

encoder_config = ViTConfig()
encoder = ViTModel(encoder_config)
model = VisionEncoderDecoderModel(encoder=encoder, decoder=decoder)
```
5. fine-tune on your custom dataset

**Comment by Jim0530** [comment]
Thanks a lot 
now training is okay.
But how do I do the inference? 
Thanks for ur help

**Comment by Jim0530** [comment]
### model
> encoder = BeitModel(BeitConfig())
decoder = TrOCRForCausalLM(TrOCRConfig(39))
model = VisionEncoderDecoderModel(encoder=encoder, decoder=decoder)

###config

>model.config.decoder_start_token_id = processor.tokenizer.cls_token_id
model.config.pad_token_id = processor.tokenizer.pad_token_id
model.config.vocab_size = model.config.decoder.vocab_size
model.config.eos_token_id = processor.tokenizer.sep_token_id
model.config.max_length = 16
model.config.early_stopping = True
model.config.no_repeat_ngram_size = 3
model.config.length_penalty = 2.0
model.config.num_beams = 4


##inference

> pixel_values = train_dataset[0]["pixel_values"].unsqueeze(0)
labels = train_dataset[0]["labels"].unsqueeze(0)
model.generate(pixel_values)

And it comes out with the error 
Error: forward() got an unexpected keyword argument 'attention_mask'

**Comment by NielsRogge** [comment]
Yes, that's because `BeitModel` currently doesn't take `attention_mask` as input to its forward method.

However, we're working on fixing this, as vision models don't actually need the attention mask.

**Comment by v3ss0n** [comment]
What if the training set is Non-English ? Bert and existing pretrained data won't work right?
